apps/meteor/client/views/admin/settings/Setting/inputs/CodeMirror/CodeMirror.tsx

```diff
@@ -44,91 +44,68 @@ function CodeMirror({
 }: CodeMirrorProps): ReactElement {
 	const [value, setValue] = useState(valueProp || defaultValue);
-
-	const textAreaRef = useRef<HTMLTextAreaElement>(null);
-	const editorRef = useRef<EditorFromTextArea | null>(null);
 	const handleChange = useEffectEvent(onChange);
 
-	useEffect(() => {
-		if (editorRef.current) {
-			return;
-		}
-
-		const setupCodeMirror = async (): Promise<void> => {
-			const { default: CodeMirror } = await import('codemirror');
-			await Promise.all([
-				import('../../../../../../../app/ui/client/lib/codeMirror/codeMirror'),
-				import('codemirror/addon/edit/matchbrackets'),
-				import('codemirror/addon/edit/closebrackets'),
-				import('codemirror/addon/edit/matchtags'),
-				import('codemirror/addon/edit/trailingspace'),
-				import('codemirror/addon/search/match-highlighter'),
-				import('codemirror/lib/codemirror.css'),
-			]);
-
-			if (!textAreaRef.current) {
-				return;
-			}
-
-			editorRef.current = CodeMirror.fromTextArea(textAreaRef.current, {
-				lineNumbers,
-				lineWrapping,
-				mode,
-				gutters,
-				foldGutter,
-				matchBrackets,
-				autoCloseBrackets,
-				matchTags,
-				showTrailingSpace,
-				highlightSelectionMatches,
-				readOnly,
-			});
-
-			editorRef.current.on('change', (doc: Editor) => {
-				const value = doc.getValue();
-				setValue(value);
-				handleChange(value);
-			});
-		};
-
-		setupCodeMirror();
-
-		return (): void => {
-			if (!editorRef.current) {
-				return;
+	const textAreaRef = useCallback(
+		async (node: HTMLTextAreaElement | null) => {
+			if (!node) return;
+
+			try {
+				const { default: CodeMirror } = await import('codemirror');
+				await Promise.all([
+					import('../../../../../../../app/ui/client/lib/codeMirror/codeMirror'),
+					import('codemirror/addon/edit/matchbrackets'),
+					import('codemirror/addon/edit/closebrackets'),
+					import('codemirror/addon/edit/matchtags'),
+					import('codemirror/addon/edit/trailingspace'),
+					import('codemirror/addon/search/match-highlighter'),
+					import('codemirror/lib/codemirror.css'),
+				]);
+
+				const editor = CodeMirror.fromTextArea(node, {
+					lineNumbers,
+					lineWrapping,
+					mode,
+					gutters,
+					foldGutter,
+					matchBrackets,
+					autoCloseBrackets,
+					matchTags,
+					showTrailingSpace,
+					highlightSelectionMatches,
+					readOnly,
+				});
+
+				editor.on('change', (doc: Editor) => {
+					const newValue = doc.getValue();
+					setValue(newValue);
+					handleChange(newValue);
+				});
+
+				return () => {
+					if (node.parentNode) {
+						editor.toTextArea();
+					}
+				};
+			} catch (error) {
+				console.error('CodeMirror initialization failed:', error);
 			}
-
-			editorRef.current.toTextArea();
-		};
-	}, [
-		autoCloseBrackets,
-		foldGutter,
-		gutters,
-		highlightSelectionMatches,
-		lineNumbers,
-		lineWrapping,
-		matchBrackets,
-		matchTags,
-		mode,
-		handleChange,
-		readOnly,
-		textAreaRef,
-		showTrailingSpace,
-	]);
-
-	useEffect(() => {
```

Comment:
after changing the setting value, saving it and reloading. The valueProps is different from what is rendered in the CodeBox. Can u check it and add a test to guarantee this?